### PR TITLE
feat(tab-bar): add expand and shape props for ionic theme

### PR DESCRIPTION
**Pull request by thetaPC**
Issue number: internal

---------

## What is the current behavior?

`ion-tab-bar` does not have the ability to change its border radius. It also can't change the width to fit its contents.

## What is the new behavior?

- Added the `shape` prop that is only used in the `ionic` theme.
- Added tests for `shape`.
- Added the `expand` prop that will determine if the tab bar will fully expand to the screen or only expand to the width of the contents.
-  `expand="compact"` (expand to the width of the contents) can only be used in the `ionic` theme.
- Added tests for `expand`.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

[Shape preview](https://ionic-framework-git-rou-11202-ionic1.vercel.app/src/components/tab-bar/test/shape?ionic:theme=ionic)
[Expand preview](https://ionic-framework-git-rou-11202-ionic1.vercel.app/src/components/tab-bar/test/expand?ionic:theme=ionic)

### CODE REVIEW

**Review by OS-giulianasilva — APPROVED** [review]

**Comment by brandyscarney** [comment]
Should this be marked as a `feat`?

File: core/src/components/tab-bar/tab-bar.tsx [hunk 1]
```diff
@@ -56,6 +56,26 @@ export class TabBar implements ComponentInterface {
...
+  @Prop() expand: 'compact' | 'full' = 'full';
+
+  /**
+   * Set to `"soft"` for a tab bar with slightly rounded corners,
+   * `"round"` for a tab bar with fully rounded corners, or
+   * `"rectangular"` for a tab bar without rounded corners.
+   *
+   * Defaults to `"round"` for the `"ionic"` theme, undefined for all other themes.
+   */
+  @Prop() shape?: 'soft' | 'round' | 'rectangular';
```
Comment (brandyscarney) [inline core/src/components/tab-bar/tab-bar.tsx:77]: Doesn't the default need to be `round` for the `ionic` theme?

**Review by brandyscarney — COMMENTED** [review]
This is looking good! Just a question on the default shape.

**Review by brandyscarney — APPROVED** [review]
Great work on this!

**Pull request merged**